compiler/lib/effects.ml

```diff
@@ -927,7 +944,7 @@ let f ~flow_info ~live_vars p =
   let cps_needed = Partial_cps_analysis.f p flow_info in
   let p, cps_needed = rewrite_toplevel ~cps_needed p in
   let p = split_blocks ~cps_needed p in
-  let p, cps_calls = cps_transform ~live_vars ~flow_info ~cps_needed p in
+  let p, trampolined_calls, in_cps = cps_transform ~live_vars ~flow_info ~cps_needed p in
```

Comment:
What's the difference between `cps_needed` and `in_cps` ?

Reply:
I’m not entirely sure, to be honest, I’d rather wait for the author @vouillon to answer.

Reply:
Basically, we turn:
```
x = f(x1, ..., xn)
cont
```
into:
```
y = f(x1, ..., xn, (x) => cont)
return y
```
when `x` is in `cps_needed`.
And then, `y` is put into `in_cps`.

You can see that we use a fresh variable here:
https://github.com/ocsigen/js_of_ocaml/blob/f47b4d5456b36e1164f428ccfa1ec4d2740a393d/compiler/lib/effects.ml#L288-L290